core/src/main/java/org/jruby/RubyProcess.java

```diff
@@ -605,12 +605,16 @@ public static IRubyObject exit_bang(IRubyObject recv, IRubyObject[] args) {
 
     @JRubyMethod(name = "groups", module = true, visibility = PRIVATE)
     public static IRubyObject groups(IRubyObject recv) {
-        long[] groups = Platform.getPlatform().getGroups(recv);
-        RubyArray ary = RubyArray.newArray(recv.getRuntime(), groups.length);
+        final Ruby runtime = recv.getRuntime();
+        long[] groups = runtime.getPosix().getgroups();
+        if (groups == null) { // not-implemented for the given platform (e.g. Windows)
+            throw runtime.newNotImplementedError("groups() function is not implemented");
```

Comment:
```suggestion
            throw runtime.newNotImplementedError("groups() function is unimplemented on this machine");
```
\* to match MRI's message